GoogleForCreators/web-stories-wp
# Design System: DropDown a11y updates

## Summary

The DropDown base PR was getting unruly so I decided to separate out the updates to accessibility to make them easier to focus on. The DropDown component in the design system aggregates a lot of the functionality (and aria) or the other custom select menus that already exist in the editor and dashboard. This PR just tests things and fixes some of the redundancies.  

We have passing AXE and lighthouse scores (the only thing either of them complained about was color contrast for the dark theme hint text, which I documented [here in my list of questions ](https://docs.google.com/document/d/1w06W9T3sxbiqveoz9Zk9_sO87JPO4seQzjo1FIDBK54/edit)for the design team after the holidays). 

## Relevant Technical Choices

- Remove redundant aria-roles on expanded "menu". This was a little tricky because sometimes we have grouped lists that create all of the options.  I see these as their own lists, all of which are under the button's given labeling which is passed to the menu's container. A watered down example would be a dropdown labeled "Pick a favorite creature", where we have a list of options grouped by type of creature. So we'd have lists of mammals, reptiles, and insects. I also did it this way because it gives our html scaffolding a less complex structure. 
- Added a memoized uuid based id to the select button to tell the menu that the select button can act as a label if needed. 
- Sets aria up in `dropDown/index` and passes to select button as `...rest` because we need the same ariaLabel vs dropDownLabel for the menu, so keeping that set up together. 

## To-do



## User-facing changes

- None, storybook only

## Testing Instructions

Run storybook locally and try using a screen reader. 
- Do you get appropriate context to use the menu? 
- You can also try setting up an 'ariaLabel' in the storybook knobs and see that that a value is spoken as opposed to the dropDownLabel (that's printed on the screen) when set. 

---



Addresses #4950
